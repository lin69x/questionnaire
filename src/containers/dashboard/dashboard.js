import React from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {Row, Col} from 'react-flexbox-grid';

import Layout from '../../components/layout';
import QuestionCard from '../../components/question-card';
import {AppActions} from '../../actions';

const Dashboard = ({
 app: {name, description, questions, currentQuestionIdx},
 prevQuestion,
 nextQuestion,
}) => (
  <Layout>
    <Col md={10} mdOffset={1}>
      <h2>{name}</h2>
      <h3>{description}</h3>
    </Col>
    <Row>
      <Col md={8} mdOffset={2}>
        <QuestionCard
          {...questions[currentQuestionIdx]}
          actions={[
            <button
              type="button"
              key='btn-prev'
              onClick={prevQuestion}
              disabled={currentQuestionIdx === 0}
            >
              Previous
            </button>,
            <button
              type="submit"
              key='btn-next'
            >
              {currentQuestionIdx === questions.length - 1 ? 'Finish' : 'Next'}
            </button>
          ]}
          onSubmit={nextQuestion}
        />
      </Col>
    </Row>
  </Layout>
);


const mapStateToProps = (state) => ({
  app: state.app
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  nextQuestion: AppActions.nextQuestion,
  prevQuestion: AppActions.prevQuestion,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
