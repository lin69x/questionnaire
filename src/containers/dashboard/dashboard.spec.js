import React from 'react';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import {Dashboard} from './dashboard';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new EnzymeAdapter() });

const props = {
  app: {
    name: 'Test Name',
    description: 'Test Description',
    questions: [],
    currentQuestionIdx: 0,
  },
};

describe('<Dashboard/>', () => {
  it('should match snapshot', () => {
    const renderedValue = renderer.create(<Dashboard { ...props }/>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
