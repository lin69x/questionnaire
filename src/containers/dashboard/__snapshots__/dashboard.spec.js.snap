// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> should match snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row center-md"
  >
    <h2>
      Test Name
    </h2>
  </div>
  <div
    className="row center-md"
  >
    <h3>
      Test Description
    </h3>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-8 col-md-offset-2"
    >
      <div
        style={
          Object {
            "background": "#fff",
            "borderRadius": 2,
            "height": 230,
            "padding": 16,
          }
        }
      >
        <div
          className="row center-md"
        >
          <h3>
            Thank You!
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
